src/scripts/ui/PlayerTradeBuildingBody.tsx

```diff
@@ -6,15 +7,49 @@ import { BuildingSellComponent } from "./BuildingSellComponent";
 import { BuildingStorageComponent } from "./BuildingStorageComponent";
 import { BuildingUpgradeComponent } from "./BuildingUpgradeComponent";
 import { BuildingWorkerComponent } from "./BuildingWorkerComponent";
+import { PendingTradesComponent } from "./PendingTradesComponent";
 import { PlayerTradeComponent } from "./PlayerTradeComponent";
 import { ResourceImportComponent } from "./ResourceImportComponent";
 
+type Tab = "trades" | "pending" | "import";
+let savedCaravanTab: Tab = "trades";
+
 export function PlayerTradeBuildingBody({ gameState, xy }: IBuildingComponentProps): React.ReactNode {
+   const [currentTab, setCurrentTab] = useState<Tab>(savedCaravanTab);
+   let content: React.ReactNode = null;
+
+   if (currentTab === "trades") {
+      savedCaravanTab = "trades";
+      content = <PlayerTradeComponent gameState={gameState} xy={xy} />;
+   } else if (currentTab === "pending") {
+      savedCaravanTab = "pending";
+      content = <PendingTradesComponent gameState={gameState} xy={xy} />;
+   } else if (currentTab === "import") {
+      savedCaravanTab = "import";
+      content = (
+         <article role="tabpanel" className="f1 column" style={{ padding: "8px" }}>
+            <ResourceImportComponent gameState={gameState} xy={xy} />
+         </article>
+      );
+   }
+
    return (
-      <div className="window-body">
+      <div className="window-body column">
          <BuildingUpgradeComponent gameState={gameState} xy={xy} key={xy} />
-         <PlayerTradeComponent gameState={gameState} xy={xy} />
-         <ResourceImportComponent gameState={gameState} xy={xy} />
+
+         <menu role="tablist">
+            <button onClick={() => setCurrentTab("trades")} aria-selected={currentTab === "trades"}>
+               Trades
```

Comment:
These needs to be added to the localization file

Reply:
Updated